File: golang/algorithm_introduction/insert_sort.go
package algorithm_introduction

type InsertSort struct {
	arr []int
}

func (is *InsertSort) Sort() {
	l := len(is.arr)
	for j := 1; j < l; j++ {
		key := is.arr[j]
		i := j - 1
		for i > 0 && is.arr[i] > key {
			is.arr[i+1] = is.arr[i]
			i = i - 1
		}
		is.arr[i+1] = key
	}
}

File: python/algorithm_introduction/insert_sort.py
from typing import List

def insert_sort(arr: List[int]) -> List[int]:
    num = len(arr)
    for j in range(2, num):
        key = arr[j]
        i = j - 1
        while i > 0 and arr[i] > key:
            arr[i + 1] = arr[i]
            i = i - 1
        arr[i + 1] = key
    return arr
